backend/generateData/generateData.js

```diff
@@ -0,0 +1,70 @@
+const mongoose = require('mongoose')
+
+require('dotenv').config()
+
+let SponsorshipFund = require('../models/SponsorshipFund.model')
+let FundingItem = require('../models/FundingItem.model')
+let UWFinancePurchase = require('../models/UWFinancePurchaseRequest.model')
+let PersonalPurchaseRequest = require('../models/PersonalPurchaseRequest.model')
+
+const dummyData = require('./dummyData')
+
+const uri = process.env.ATLAS_URI
+mongoose.connect(uri, {
+    useNewUrlParser: true,
+    useUnifiedTopology: true,
+})
+
+// that automatically creates (at least 2) sponsorship funds, with associated funding items, and associated uw finance purchases and personal purchases.
+
+const connection = mongoose.connection
+
+connection.once('open', () => {
```

Comment:
must await connection before writing, otherwise it will break if your connection takes time to open